### ISSUE TITLE
false negative Path::getParent

### ISSUE THREAD

**Issue by xenoterracide** [uber/NullAway#463]
`Path::getParent` can return null if you are at `/`

```java
  @Test
  void test() {
    findConfig( Path.of( "" ) );
  }

  Optional<Path> findConfig( Path searchDir ) {
    var configFile = searchDir.resolve("foo.yml");

    if ( Files.exists( configFile ) ) {
      return Optional.of( configFile );
    }

    return this.findConfig( searchDir.getParent() );
  }
```

here's the relevant portion of my config, it's a `buildSrc` kotlin plugin but that shouldn't be relevant.

```kts
plugins {
  `java-library`
  `java-test-fixtures`
  id("brix.bom")
  id("net.ltgt.errorprone")
  id("org.checkerframework")
}

val checker = the<LibrariesForChecker>()

dependencies {
  errorprone("com.google.errorprone:error_prone_core:2.4.+")
  checkerFramework(checker.processor)
  compileOnly(checker.annotations)
  testFixturesCompileOnly(checker.annotations)
}

checkerFramework {
  excludeTests = true
  if (Files.exists(buildDir.toPath().resolve("../config/stubs"))) {
    extraJavacArgs.addAll(listOf("-Astubs=$buildDir/../config/stubs"))
  }
  checkers.addAll(
    listOf(
      "org.checkerframework.checker.nullness.NullnessChecker"
    )
  )
}
```

```
\--- com.uber.nullaway:nullaway:0.8.+ -> 0.8.0
     \--- com.google.guava:guava:22.0 -> 30.1-jre (*)

```

```
------------------------------------------------------------
Gradle 7.0
------------------------------------------------------------

Build time:   2021-04-09 22:27:31 UTC
Revision:     d5661e3f0e07a8caff705f1badf79fb5df8022c4

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.10 (AdoptOpenJDK 11.0.10+9)
OS:           Linux 5.11.10-200.fc33.x86_64 amd64
```

```
Fedora 33
5.11.10-200.fc33.x86_64
 09:47:23 up 26 days, 10:53,  1 user,  load average: 1.82, 1.38, 1.18
              total        used        free      shared  buff/cache   available
Mem:            15G         11G        1.3G        1.3G        3.4G        3.0G
Swap:           12G        2.6G        9.4G
```

**Comment by lazaroclapp** [comment]
What's the fully qualified name of `Path` here? Is this `java.nio.file.Path`?

Note that for third-party libraries (including JDK libraries), which aren't sources being analyzed by NullAway, the tool will make optimistic assumptions (assume the method can take nulls everywhere and returns non-null). So, this sort of behavior is not unexpected, and can be resolved on your end by adding a custom [library model](https://github.com/uber/NullAway/wiki/Configuration#library-models).

That said, happy to look into including `java.nio.file.Path` in our [default models](https://github.com/uber/NullAway/blob/master/nullaway/src/main/java/com/uber/nullaway/handlers/LibraryModelsHandler.java#L538), since it's certainly a frequently used JDK class and the issue you are seeing is likely common to most NullAway users! 

Will look into this soon (hopefully within the week), and happy to accept a PR if you/anyone else can get to it before I can 🙂

**Comment by lazaroclapp** [comment]
Actually, got a chance to do this today. Model for `Path.getParent()` should be part of the next NullAway release!

**Comment by xenoterracide** [comment]
Thanks and yes it's the jdk Path
